spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,199 @@
+      reservation1
+      reservation2
+      block1
+      block2
+      start_date = "march 14, 2019"
+      end_date = "march 20, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms.length).must_equal 13
+    end
+
+    it "can return available rooms for a specified date range, full range unavail" do
+      reservation1
+      reservation2
+      block1
+
+      start_date = "march 17, 2019"
+      end_date = "march 20, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms.length).must_equal 13
+    end
+
+    it "can return available rooms for a specified date range, start_date unavail" do
+      reservation1
+      reservation2
+      block1
+      block2
+
+      start_date = "march 20, 2019"
+      end_date = "march 25, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms.length).must_equal 10
+    end
+
+    it "can return an empty array when no rooms are available for a specified date range" do
+      Hotel::ReservationManager::MAX_ROOMS.times do
+        manager.make_reservation("march 15, 2019", "march 20, 2019")
+      end
+
+      start_date = "march 15, 2019"
+      end_date = "march 16, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms).must_be_empty
+    end
+
+    it "includes all rooms if no reservations" do
+      start_date = "march 14, 2019"
+      end_date = "march 16, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms.length).must_equal 20
+    end
+
+    it "does not include the booked rooms" do
+      reservation1
+      reservation2
+      start_date = "march 14, 2019"
+      end_date = "march 16, 2019"
+      vacant_rooms = manager.available_rooms(start_date, end_date)
+      expect(vacant_rooms.length).must_equal 19
+    end
+  end
+
+  describe "make_block method" do
+    it "Can reserve a block of 5 rooms from available rooms" do
+      reservation1
+      reservation2
+      new_block = manager.make_block(5, "march 15, 2019", "march 20, 2019", 1)
+      vacant_rooms = manager.available_rooms("march 15, 2019", "march 20, 2019")
+      expect(new_block.length).must_equal 5
+      expect(vacant_rooms.length).must_equal 13
+    end
+
+    it "Will not show blocked rooms as available" do
+      reservation1
+      reservation2
+      manager.make_block(4, "march 19, 2019", "march 23, 2019", 1)
+
+      vacant_rooms = manager.available_rooms("march 19, 2019", "march 20, 2019")
+      expect(vacant_rooms.length).must_equal 14
+    end
+
+    it "Raises and argument error for room blocks with more than 5 rooms" do
+      expect { manager.make_block(6, "march 19, 2019", "march 23, 2019", 1) }.must_raise ArgumentError
+    end
+  end
+
+  describe "check_block_availability method" do
+    it "Will show rooms still available in a specified block" do
```

Comment:
I love this test! I could definitely imagine it being tidied up so it's completely clear what's going on. I almost would like this test a lot if it were like this:

```ruby
start_date = "march 19, 2019"
      end_date = "march 23, 2019"
      id = 1
      num_of_blocked_rooms = 5
      manager.make_block(num_of_blocked_rooms, start_date, end_date, id)
      expect(manager.reservations_by_date(start_date, end_date).length).must_equal 0
      expect(manager.blocked_rm_by_date(start_date, end_date).length).must_equal num_of_blocked_rooms

      reserved_room = manager.reserve_block_room(id)
      expect(manager.reservations_by_date(start_date, end_date).length).must_equal 1
      expect(reserved_room[0].block_id).must_equal 1
      expect(manager.blocked_rm_by_date(start_date, end_date).length).must_equal (num_of_blocked_rooms - 1)
```